webapp/src/marked.tsx

```diff
@@ -0,0 +1,148 @@
+
+import * as React from "react";
+import * as ReactDOM from "react-dom";
+import * as data from "./data";
+import * as marked from "marked";
+
+type ISettingsProps = pxt.editor.ISettingsProps;
+
+interface MarkedContentProps extends ISettingsProps {
+    markdown: string;
+}
+
+interface MarkedContentState {
+}
+
+export class MarkedContent extends data.Component<MarkedContentProps, MarkedContentState> {
+
+    // Local cache for images, cleared when we create a new project.
+    // Stores code => data-uri image of decompiled result
+    private static blockSnippetCache: pxt.Map<string> = {};
+    public static clearBlockSnippetCache() {
+        this.blockSnippetCache = {};
+    }
+
+    // This should probably be moved into some shared util between docsrenderer and the webapp
+    private replaceAll(replIn: string, x: string, y: string) {
+        return replIn.split(x).join(y)
+    }
+
+    private htmlQuote(s: string): string {
+        s = this.replaceAll(s, "&", "&amp;")
+        s = this.replaceAll(s, "<", "&lt;")
+        s = this.replaceAll(s, ">", "&gt;")
+        s = this.replaceAll(s, "\"", "&quot;")
+        s = this.replaceAll(s, "\'", "&#39;")
+        return s;
+    }
+
+    // the input already should be HTML-quoted but we want to make sure, and also quote quotes
+    private html2Quote(s: string) {
```

Comment:
do we still need this help? Why not just htmlEscape?

Reply:
I was just following what was done in the docs, but perhaps not.